src/main/java/com/tmh/controller/admin/OrderController.java

```diff
@@ -0,0 +1,131 @@
+package com.tmh.controller.admin;
+
+import java.util.Locale;
+
+import javax.validation.Valid;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import com.tmh.entities.Order;
+import com.tmh.entities.OrderItem;
+
+@Controller
+public class OrderController extends AdminController {
+
+	@RequestMapping(value = "/orders")
+	public String showOrderList(Model model) {
+		logger.info("show orders list");
+		
+		loadModelAttribute(model);
+		
+		return "views/admin/orderManager/orderList";
+	}
+	
+	@RequestMapping(value = "/orders/{id}")
+	public String showOrderDetail(@PathVariable("id") int id, Model model) {
+		logger.info("show order detail");
+		
+		Order order = orderService.findById(id);
+		
+		if (order == null) {
+			model.addAttribute("css", "error");
+			model.addAttribute("msg", messageSource.getMessage("order.not.found", null, Locale.US));
+		}
```

Comment:
nếu order bị null thì em return về page list hoặc page lỗi luôn chứ add code ở line 45 chị thất hơi sai

Reply:
Dạ order null thì em vẫn trả ra trang detail ạ, nhưng ở trang đó em để 1 cái alert là Order Not Found chị à!!

Reply:
đã ko có thì trả về page detail nó vô nghĩa ko nhỉ, nên trả về page list order để ng ta chọn cái order khác thì ổn hơn